sensor/common/scannerdefinitions/http_handler.go

```diff
@@ -20,29 +22,48 @@ var (
 	headersToProxy = set.NewFrozenStringSet("If-Modified-Since", "Accept-Encoding")
 )
 
-// scannerDefinitionsHandler handles requests to retrieve scanner definitions
+// ScannerDefinitionHandler handles requests to retrieve scanner definitions
 // from Central.
-type scannerDefinitionsHandler struct {
-	centralClient *http.Client
+type ScannerDefinitionHandler struct {
+	centralClient    *http.Client
+	centralReachable *atomic.Bool
 }
 
 // NewDefinitionsHandler creates a new scanner definitions handler.
-func NewDefinitionsHandler(centralEndpoint string) (http.Handler, error) {
+func NewDefinitionsHandler(centralEndpoint string) (*ScannerDefinitionHandler, error) {
 	client, err := clientconn.NewHTTPClient(mtls.CentralSubject, centralEndpoint, 0)
 	if err != nil {
 		return nil, errors.Wrap(err, "instantiating central HTTP transport")
 	}
-	return &scannerDefinitionsHandler{
-		centralClient: client,
+	return &ScannerDefinitionHandler{
+		centralClient:    client,
+		centralReachable: &atomic.Bool{},
 	}, nil
 }
 
-func (h *scannerDefinitionsHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+// Notify reacts to sensor going into online/offline mode.
```

Comment:
does Sensor have a separate offline mode from the product itself? If not, why can't Sensor get this data from Central?

Reply:
Offline mode is when Sensor gets disconnected from Central. This information comes from the Sensor gRPC component when it identifies that the gRPC stream was closed.

Reply:
gotcha thanks. Offline mode sounds misleading because it makes it sound like it was done on purpose. Is it or is just just a lost connection?

Reply:
It's just a lost connection. The feature is not yet released. There is still time to rename it :-)